tests/unit/app/test_segments.py

```diff
@@ -18,29 +18,35 @@ def test_segments(df_lending_club, tmpdir):
     writer_config = WriterConfig("local", ["protobuf"], output_path.realpath())
     yaml_data = writer_config.to_yaml()
     WriterConfig.from_yaml(yaml_data)
+    test_segments = [
+        [{"key": "home_ownership", "value": "RENT"}],
+        [{"key": "home_ownership", "value": "MORTGAGE"}],
+    ]
 
     session_config = SessionConfig("project", "pipeline", writers=[writer_config])
     with session_from_config(session_config) as session:
         with session.logger(
             "test",
-            segments=[
-                [{"key": "home_ownership", "value": "RENT"}],
-                [{"key": "home_ownership", "value": "MORTGAGE"}],
-            ],
+            segments=test_segments,
             cache_size=1,
         ) as logger:
             logger.log_dataframe(df_lending_club)
             profile = logger.profile
             profiles = logger.segmented_profiles
-            mortage_segment = logger.get_segment([{"key": "home_ownership", "value": "MORTGAGE"}])
+            mortage_segment = logger.get_segment(test_segments[1])
 
     assert profile is None
     assert len(profiles) == 2
-    assert profiles[list(profiles.keys())[0]].tags["segment"] == json.dumps([{"key": "home_ownership", "value": "RENT"}])
-    assert profiles[list(profiles.keys())[1]].tags["segment"] == json.dumps([{"key": "home_ownership", "value": "MORTGAGE"}])
+    segment_keys = [key for key in profiles[list(profiles.keys())[0]].tags.keys() if key.startswith(_TAG_PREFIX)]
+    for segment_key in segment_keys:
+        assert profiles[list(profiles.keys())[0]].tags[segment_key] == test_segments[0][0][_TAG_VALUE]  # 'RENT'
```

Comment:
nit: it's worth having a few explicit assertions here (assert `whylogs.tag.key=RENT`